### PR TITLE
feat: Comprehensive update with documentation, Node.js integration, a…

### PR DESCRIPTION
**Pull request by dirvine**
# Node.js Integration, Documentation Enhancement, and AttoTokens Fix

## Overview
This PR introduces comprehensive documentation, Node.js integration capabilities, and fixes a critical issue with AttoTokens parsing. The changes aim to improve developer experience, expand platform accessibility, and enhance system reliability.

### Documentation Enhancements
- Added MkDocs-based documentation system with comprehensive structure
- Implemented GitHub Actions workflow for automatic documentation deployment
- Created detailed guides for:
  * Client modes and usage
  * Data storage and management
  * EVM integration
  * Local development setup
  * Testing procedures
  * Payment handling
- Added API documentation for:
  * ant-node
  * autonomi-client
  * blsttc
  * self-encryption

### Node.js Integration
- Added Node.js bindings for core functionality:
  * Linked List operations
  * Pointer management
  * Vault interactions
  * Wallet operations
- Included TypeScript definitions for improved type safety
- Added comprehensive tests for Node.js bindings
- Created local testnet action for testing Node.js integrations
- Added Node.js-specific documentation and examples

### Bug Fixes
- Fixed AttoTokens parsing to properly handle large values:
  * Added validation for excessive values before multiplication
  * Updated precision handling for decimal places
  * Enhanced error handling for edge cases
  * Added comprehensive test coverage for token parsing

### Infrastructure Improvements
- Updated `.gitignore` to handle Node.js and documentation artifacts
- Added GitHub Actions workflow for documentation
- Updated dependency management in Cargo.lock

## Testing
- All Rust tests are passing, including the enhanced AttoTokens tests
- Added Node.js integration tests
- Documentation build and deployment verified
- Local testnet functionality verified

## Impact
These changes significantly improve the project's accessibility and reliability:
- Developers can now integrate with the system using Node.js
- Documentation is more comprehensive and easier to maintain
- Token handling is more robust and reliable
- Testing infrastructure is more comprehensive

## Breaking Changes
None. All changes are backward compatible.

## Future Work
- Consider adding more Node.js examples
- Expand documentation with more use cases
- Add performance benchmarks for Node.js bindings

## Checklist
- [x] Documentation updated
- [x] Tests added and passing
- [x] Code follows project style guidelines
- [x] CI/CD pipelines updated
- [x] No breaking changes introduced

### CODE REVIEW

File: docs/online-documentation/api/autonomi-client/README.md [hunk 1]
```diff
@@ -0,0 +1,654 @@
...
+    };
+    const client = await Client.initWithConfig(config);
+    ```
+
+=== "Python"
+    ```python
+    from autonomi import Client
+
+    # Initialize a read-only client
+    client = Client.init_read_only()
```
Comment (grumbach) [inline docs/online-documentation/api/autonomi-client/README.md:51]: As we got rid of the wallet change, maybe we should update this accordingly?
